### PR TITLE
Dependencies to build Webkit

### PR DESCRIPTION
**Pull request by ilhooq**
I experiment to cross-build [WebKit](https://github.com/WebKit/WebKit) ([WinCairo](https://trac.webkit.org/wiki/BuildingCairoOnWindows) port) using mxe but they are some missing libraries : [libpsl](https://github.com/rockdaboot/libpsl), [woff2](https://github.com/google/woff2), [brotli](https://github.com/google/brotli).

This is a proposal to add these libraries in the src tree.

### CODE REVIEW

File: src/brotli.mk [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+$(PKG)_IGNORE   :=
+$(PKG)_VERSION  := 1.0.9
+$(PKG)_CHECKSUM := f9e8d81d0405ba66d181529af42a3354f838c939095ff99930da6aa9cdf6fe46
+$(PKG)_GH_CONF  := google/brotli/releases/tag,v,,
+$(PKG)_DEPS     := cc
+
+define $(PKG)_BUILD
+    cd '$(BUILD_DIR)' && '$(TARGET)-cmake' \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_INSTALL_PREFIX='$(PREFIX)/$(TARGET)' \
```
Comment (tonytheodore) [inline src/brotli.mk:14]: CMAKE_BUILD_TYPE and CMAKE_INSTALL_PREFIX are set by the toolchain file and should be removed

File: src/libpsl.mk [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+$(PKG)_DEPS     := cc libidn2 libunistring
+
+define $(PKG)_BUILD
+    cd '$(BUILD_DIR)' && '$(SOURCE_DIR)/configure' \
+    --host='$(TARGET)' \
+    --prefix='$(PREFIX)/$(TARGET)' \
+    $(if $(BUILD_STATIC), \
+        --enable-static --disable-shared , \
+        --disable-static --enable-shared ) \
+    --libdir='$(PREFIX)/$(TARGET)/lib'
```
Comment (tonytheodore) [inline src/libpsl.mk:18]: does $(MXE_CONFIGURE_OPTS) cover all of those?

File: src/woff2.mk [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+define $(PKG)_BUILD
+    cd '$(BUILD_DIR)' && '$(TARGET)-cmake' \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DGEN_PKG_VERSION=ON \
+        $(if $(BUILD_STATIC), \
+            -DBUILD_SHARED_LIBS=OFF , \
+            -DBUILD_SHARED_LIBS=ON ) \
+        -DCMAKE_INSTALL_PREFIX='$(PREFIX)/$(TARGET)' \
+        '$(SOURCE_DIR)'
```
Comment (tonytheodore) [inline src/woff2.mk:19]: only GEN_PKG_VERSION is required, rest set by toolchain file

**Comment by tonytheodore** [comment]
Can you rebase and squash this?

**Comment by ilhooq** [comment]
I rebased the code and created new commits in front of your master branch. I think it should be clean now to merge my changes.

**Comment by tonytheodore** [comment]
Thanks!

**Comment by tonytheodore** [comment]
I had to merge it manually.

**Pull request closed without merging**